lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -464,6 +464,19 @@ def connect_to_machine(name, chef_server = nil)
       machine_for(machine_spec, machine_spec.reference)
     end
 
+    def stop_machine(action_handler, machine_spec, machine_options)
+      instance = instance_for(machine_spec)
+      if instance && instance.exists?
```

Comment:
We'd had some conversations elsewhere about the idea of "exists" having a variety of meanings where AWS instances are concerned, and using a different set of query functions to determine the exact state needed.